### PR TITLE
build rtabmap_viz with Qt5 on Ubuntu 22.10

### PR DESCRIPTION
**Pull request by lucasw**
I just tried building latest on Ubuntu 22.10 and got

```
In file included from /home/lucasw/base_catkin_ws/src/other/rtabmap_ros/rtabmap_viz/include/rtabmap_viz/PreferencesDialogROS.h:32,
                 from /home/lucasw/base_catkin_ws/src/other/rtabmap_ros/rtabmap_viz/src/PreferencesDialogROS.cpp:28:
/home/lucasw/base_catkin_ws/devel/include/rtabmap-0.21/rtabmap/gui/PreferencesDialog.h:33:10: fatal error: QDialog: No such file or directory
   33 | #include <QDialog>
      |          ^~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/rtabmap_viz.dir/build.make:104: CMakeFiles/rtabmap_viz.dir/src/PreferencesDialogROS.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
/home/lucasw/base_catkin_ws/src/other/rtabmap_ros/rtabmap_viz/src/GuiNode.cpp:31:10: fatal error: QApplication: No such file or directory
   31 | #include <QApplication>
      |          ^~~~~~~~~~~~~~
compilation terminated.
/home/lucasw/base_catkin_ws/src/other/rtabmap_ros/rtabmap_viz/src/GuiWrapper.cpp:29:10: fatal error: QApplication: No such file or directory
   29 | #include <QApplication>
      |          ^~~~~~~~~~~~
```

I added `find_package(Qt5 COMPONENTS Widgets REQUIRED)` and `include_directories(${Qt5Widgets_INCLUDE_DIRS})` but got this:

```
/usr/include/x86_64-linux-gnu/qt5/QtCore/qglobal.h:1281:4: error: #error "You must build your code with position independent code if Qt was built with -reduce-relocations. " "Compile your code with -fPIC (and not with -fPIE)."
 1281 | #  error "You must build your code with position independent code if Qt was built with -reduce-relocations. "\
```

Adding `target_compile_options(rtabmap_viz PRIVATE -fPIC)` looks to work to fix that, the application runs:

```
rosrun rtabmap_viz rtabmap_viz
```

I'm guessing the same qt4 + qt5 support in rtabmap_rviz_plugins/CMakeLists.txt is needed here instead of just qt5

### CODE REVIEW

**Comment by lucasw** [comment]
Also there's a circular dependency that results in this (unhlepful) error:

```
Starting  >>> rtabmap_rviz_plugins                                                                                                                                                                                
[build] An internal error occurred!                                                                                                                                                                               
Traceback (most recent call last):
  File "/home/lucasw/other/install/lib/python3.10/site-packages/catkin_tools/verbs/catkin_build/build.py", line 564, in build_isolated_workspace
    all_succeeded = run_until_complete(execute_jobs(
  File "/home/lucasw/other/install/lib/python3.10/site-packages/catkin_tools/execution/executor.py", line 378, in run_until_complete
    return loop.run_until_complete(coroutine)
  File "/usr/lib/python3.10/asyncio/base_events.py", line 646, in run_until_complete
    return future.result()
  File "/home/lucasw/other/install/lib/python3.10/site-packages/catkin_tools/execution/executor.py", line 276, in execute_jobs
    done_job_fs, active_job_fs = await asyncio.wait(
  File "/usr/lib/python3.10/asyncio/tasks.py", line 368, in wait
    raise ValueError('Set of coroutines/Futures is empty.')
ValueError: Set of coroutines/Futures is empty.
```

I put another commit on that removes that, though `catkin_lint` reveals a lot of other dependency warnings, which maybe can be ignored for now but become issues later.

**Comment by matlabbe** [comment]
I didn't see the qt error on Ubuntu 22.04 / ROS humble: https://github.com/introlab/rtabmap_ros/actions/runs/4333862313/jobs/7567343743, or on ubuntu 20.04 / noetic using also qt5: https://github.com/introlab/rtabmap_ros/actions/runs/4333862427/jobs/7567343892

Normally, Qt should be included indirectly from `rtabmap_gui` target (see [here](https://github.com/introlab/rtabmap/blob/98635ae33d617afb2aa2b1f37434ffd7826640cd/RTABMapConfig.cmake.in#L10)), which should be included by `rtabmap_conversions` (included by `rtabmap_sync` in `rtabmap_viz`):
https://github.com/introlab/rtabmap_ros/blob/edd86e4f56934980f860a6c4e4b479fdb1110e94/rtabmap_conversions/CMakeLists.txt#L18

For all the refactoring of the package.xml, thanks! For ROS1 on ubuntu 22.04, is your docker image from https://github.com/lucasw/ros_from_src available directly from dockerhub? Otherwise, I am rebuilding one now to test your PR here, and I could put this container on our dockerhub so that I can add it to my ros1 docker workflows.

**Comment by matlabbe** [comment]
After reproducing the Qt error in your docker file, I've found it is related to `catkin build` (with `catkin_make` it doesn't show up). The error is maybe upstream in rtabmap library, catkin may not use cmake targets (new cmake way) but libraries and include directories (old cmake way), thus as rtabmap targets will only provide `Qt5::` cmake targets, the include may be missing downstream. I'll check how to fix this.

**Comment by matlabbe** [comment]
The problem is that `catkin build` doesn't use cmake targets for dependencies. To get `-fPIC` compilation flags from Qt, normally we would get it from the Qt5 cmake target. On ros2, `colcon build` does use cmake targets, it is why I didn't catch that issue before. 

I didn't realize before that `catkin build` compiles in isolation all packages, thus includes or compilation definitions are not shared to other packages like with `catkin_make`. I will change CI to use `catkin build` instead to catch earlier those issues (EDIT: issue added here https://github.com/introlab/rtabmap_ros/issues/906). `catkin_make` was also not able to detect cycling dependencies in `package.xml` like what you fixed in this MR. 

For the Qt include directories, I fixed it upstream: https://github.com/introlab/rtabmap/commit/e51d161407ea4f02ce5f4dcddb7046c90d71f27b

In summary, thank you for this MR! it will make it easier when I will update rosdistro.

cheers,
Mathieu
